app/src/main/java/com/example/cartoon/Sp_screen.kt

```diff
@@ -1,33 +1,28 @@
 package com.example.cartoon
 
-import android.animation.ObjectAnimator
 import android.content.Intent
 import android.os.Bundle
-import android.os.Handler
-import android.view.animation.AccelerateDecelerateInterpolator
-import android.view.animation.DecelerateInterpolator
-import android.widget.ImageView
 import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
+import com.example.cartoon.databinding.ActivitySpScreenBinding
 
 class Sp_Screen : AppCompatActivity() {
+    private lateinit var binding : ActivitySpScreenBinding
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_sp_screen)
+        binding = ActivitySpScreenBinding.inflate(layoutInflater)
+        setContentView(binding.root)
 
-        val textView = findViewById<TextView>(R.id.textView);
         val ThinkBot = "CARTOON!"
-        val stringBuilder1 = StringBuilder()
-        val t1 = Thread {
+        Thread {
             for (letters in ThinkBot) {
-                stringBuilder1.append(letters)
-                Thread.sleep(250)
+                StringBuilder().append(letters)
+                Thread.sleep(250)                                  //0.25 sec for each letter
                 runOnUiThread {
-                    textView.text = stringBuilder1.toString()
+                    binding.txtTitle.text = StringBuilder().toString()
```

Comment:
I believe, the txtTile.text will always be empty because of the new StringBuilder() object. you can keep the previous change of StringBuilder or define a new StringBuilder inside Thread.

Reply:
Noted sir.